openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -392,6 +394,17 @@ def validate_user_preference_serializer(serializer, preference_key, preference_v
                 "user_message": user_message,
             }
         })
+    if preference_key == "time_zone" and preference_value not in common_timezones_set:
+        developer_message = ugettext_noop(u"Value '{preference_value}' not valid for preference '{preference_key}': Not in timezone set.")  # pylint: disable=line-too-long
+        user_message = ugettext_noop(u"Value '{preference_value}' is not valid for user preference '{preference_key}'.")
```

Comment:
@catong I changed the message to be what you specified above. I'll push the updated code in PR #12870 so that we can close this one.